kvbc/test/v4blockchain/v4_latest_keys_test.cpp

```diff
@@ -151,6 +152,244 @@ TEST(column_family, set_filter) {
   }
 }
 
+TEST_F(v4_kvbc, throw_on_adding_overlap_to_range_delete) {
```

Comment:
Would there ever be situation where blocks with markers get pruned before applying range-delete? Do we need such tests?